integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -142,6 +142,11 @@ struct TestReshardingParameters {
     // catching up by the end of the epoch, and then misses a chunk. This can be fixed by using a longer
     // epoch length, but it's good to also check what happens with shorter ones.
     all_chunks_expected: bool,
+    /// When there's a resharding task delay and single-shard tracking, the delay might be pushed out
+    /// even further because the resharding task might have to wait for the state snapshot to be made
+    /// before it can proceed, which might mean that flat storage won't be ready for the child shard for a whole epoch.
+    /// This is fine and shouldn't be representative of what happens in practice
+    ready_flat_storage_expected_mid_epoch: bool,
```

Comment:
What if instead of this flag we add a way to make epochs longer?

Reply:
Yeah why not. This actually exposed a bug here. If you set the epoch length to 10, the test fails with a lot of `Can't get trie nodes for state part` errors. It doesn't really look to me like the bug is in this PR really, but the state root we get from [this flat state iter](https://github.com/near/nearcore/blob/beb00f4aa1b344a196012b9357b02e21cf167aab/core/store/src/trie/state_parts.rs#L247) does not match the expected state root, and then we get a `MissingTrieValue` error when trying to look up the state root [here](https://github.com/near/nearcore/blob/beb00f4aa1b344a196012b9357b02e21cf167aab/core/store/src/trie/state_parts.rs#L323). It's weird because the flat state has all the same values as the memtrie, so theres some bug in that state part generation code somewhere

Reply:
I see your point, yeah might be better to add a new `iter_range_with_changes` to not modify existing behavior. But it's another story.

Reply:
actually I was wrong here, it wasnt because of that. It was because if catchup finished at a height <= the wanted state snapshot, we would finish catchup and call `create_flat_storage_for_shard()` (which makes sense bc it's at a height <= to the one we want to snapshot at, so should be all good to finish). But then the `move_head_enabled` was `true` for that newly created flat storage even tho it's false for all the other ones bc we still havent made a snapshot yet. then that flat storage gets updated during chunk processing bc we should have set `move_head_enabled` to false.

fixed that one, but theres actually still a failure if u set epoch length to 10, I thnk bc `set_state_finalize()` sets flat storage status to ready even if catchup is not finished, and then we expected to read a `Resharding(Catchup)` status and panic. Will fix in another PR

Reply:
FYI I plan to land some changes that will make changing epoch length easier in the testloop